MakeHist/mumu.C

```diff
@@ -162,15 +154,36 @@ void mumu::Loop(TString sample_name)
                                        //Number of b-jet
                                        Nbjet ++;
                                    }
+                                   else{
+                                       if(jet_pt[NJ] > 30){//non_bjet pT > 30
+                                           //non b-jet pT and eta
+                                           NonbjetpT_hist->Fill(jet_pt[NJ]);
+                                           Nonbjeteta_hist->Fill(jet_eta[NJ]);
+                                           //Save non b-jet information
+                                           TLorentzVector non_bJet;
+                                           non_bJet.SetPtEtaPhiE(jet_pt[NJ],jet_eta[NJ],jet_phi[NJ],jet_E[NJ]);
+                                           non_bJets.push_back(non_bJet);
+
+                                           //Number of non b-jet
+                                           Nnon_bjet ++;
+                                       }
+                                   }
                                }
                            }
                        }// b jet pT > 20 GeV , non-b jet pT > 30 GeV
 
-                       njet_hist->Fill(Njet);
-                       if(Njet < 2) continue;//Number of jet >=2
-                       
                        nbjet_hist->Fill(Nbjet);
-                       if(Nbjet < 1) continue;// >= 1 b-jet
+                       nNonbjet_hist->Fill(Nnon_bjet);
+                       njvsnbj_hist->Fill(Nnon_bjet,Nbjet);
+
+                       if(Nbjet != 1 || Nnon_bjet < 1) continue;//N_bjet = 1, N_jet >= 2
```

Comment:
You should have one if clause for each selection here, given that the cuts for each will diverge.
Switching manually between either selection is error-prone and should be fixed.

Reply:
Also, which selection is this supposed to be? Requiring exactly 1 bottom-tagged jet and at least 1 jet that is not tagged is none of the selection categories you showed last meeting.

Reply:
As of now, the code that follows is ineffective, as this statement skips any event with other than 1 bottom-tagged jet. So your Nbjet>1 case in line 246 should never trigger.

Reply:
For example, I could not figure out how to define this region with a single 'if' function, when Case 1 (N_jet> = 2, N_bjet> = 1). So I created two output files using two 'if' function and I combined it up using 'hadd'. (N_jet >= 2, N_bjet = 1  // N_jet >= 2, N_bjet >= 2)
But in your comments, you learned how to define an area using an 'if' function.
Your comments have been a great help! Thank you.

I will update the codes after fixing this.